File: Lib/Telerik/Examples/QuickStart/Editors/TimeSpanPicker/FirstLook/MovieDurationTimeSpanPickerComponent.vb
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.Editors.TimeSpanPicker.FirstLook
    Public Class MovieDurationTimeSpanPickerComponent
        Inherits MinuteTimeSpanPickerComponent

        Public Sub New(ByVal owner As ITimeSpanPickerContentElement)
            MyBase.New(owner)
        End Sub

        Protected Overrides Sub InitializeFields()
            MyBase.InitializeFields()
            Me.DefaultSize = New System.Drawing.Size(120, 200)
        End Sub

        Public Overrides Function CreateVisualElement() As BaseTimeSpanPickerUIComponent
            Return New MovieDurationTimeSpanPickerUIComponent(Me)
        End Function
    End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/Editors/TimeSpanPicker/FirstLook/MovieDurationTimeSpanPickerComponent.cs
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.Editors.TimeSpanPicker.FirstLook
{
    public class MovieDurationTimeSpanPickerComponent : MinuteTimeSpanPickerComponent
    {
        public MovieDurationTimeSpanPickerComponent(ITimeSpanPickerContentElement owner)
            : base(owner)
        {
        }

        protected override void InitializeFields()
        {
            base.InitializeFields();
            this.DefaultSize = new System.Drawing.Size(120, 200);
        }

        public override BaseTimeSpanPickerUIComponent CreateVisualElement()
        {
            return new MovieDurationTimeSpanPickerUIComponent(this);
        }
    }
}